CODE_STYLE.md

```diff
@@ -0,0 +1,57 @@
+## Code style
+> [!IMPORTANT]  
+> All classes do live in the `godot` namespace   
+> Your linter must be compatible with code style defined in `.clang-format`
+### Naming
+1. Function naming: `snake_case`
+2. Variable naming: `snake_case`
+3. Class naming: `PascalCase`
+4. Parameter naming: `snake_case` with `p_` as an prefix
```

Comment:
4. to discuss/disagree from my point of view. We should avoid global variables, then issue with class fields prefix disappear. Alternatively, we can use `this->foo` instead of `p_foo`. When we introduce syntax prefix, we may end with more unnecessary prefixes related to language syntax instead.

Reply:
this is just the convention we are familiar with when working with godot c++ files, like this https://github.com/godotengine/godot-cpp/blob/master/src/core/method_bind.cpp#L63
assumption: other godot devs will know this already

Advantages:
* readability: you *see* which value is a parameter without knowing it's semantics (you don't need to remember or check what the value is for)
* consistency: we can skip `this->` everywhere (alternatively we should use `this->` everywhere)
* shadowing avoidance by convention

Reply:
Ok, convention is better than lack of it, if Godot uses prefixes, we may too. +1 from me.